server/src/main/java/org/opensearch/gateway/remote/ClusterStateDiffManifest.java

```diff
@@ -0,0 +1,635 @@
+                    case TO_STATE_UUID_FIELD:
+                        builder.toStateUUID(parser.text());
+                        break;
+                    case CLUSTER_BLOCKS_UPDATED_FIELD:
+                        builder.clusterBlocksUpdated(parser.booleanValue());
+                        break;
+                    case DISCOVERY_NODES_UPDATED_FIELD:
+                        builder.discoveryNodesUpdated(parser.booleanValue());
+                        break;
+                    default:
+                        throw new XContentParseException("Unexpected field [" + currentFieldName + "]");
+                }
+            } else {
+                throw new XContentParseException("Unexpected token [" + token + "]");
+            }
+        }
+        return builder.build();
+    }
+
+    @Override
+    public String toString() {
+        return Strings.toString(MediaTypeRegistry.JSON, this);
+    }
+
+    public List<String> findRemovedIndices(Map<String, IndexMetadata> indices, Map<String, IndexMetadata> previousIndices) {
+        List<String> removedIndices = new ArrayList<>();
+        for (String index : previousIndices.keySet()) {
+            // index present in previous state but not in current
+            if (!indices.containsKey(index)) {
+                removedIndices.add(index);
+            }
+        }
+        return removedIndices;
+    }
+
+    private static List<String> convertListToString(List<Object> list) {
+        List<String> convertedList = new ArrayList<>();
+        for (Object o : list) {
+            convertedList.add(o.toString());
+        }
+        return convertedList;
+    }
+
+    public List<String> findUpdatedIndices(Map<String, IndexMetadata> indices, Map<String, IndexMetadata> previousIndices) {
+        List<String> updatedIndices = new ArrayList<>();
+        for (String index : indices.keySet()) {
+            if (!previousIndices.containsKey(index)) {
+                updatedIndices.add(index);
+            } else if (previousIndices.get(index).getVersion() != indices.get(index).getVersion()) {
+                updatedIndices.add(index);
+            }
+        }
+        return updatedIndices;
+    }
+
+    public List<String> getIndicesRoutingDeleted(RoutingTable previousRoutingTable, RoutingTable currentRoutingTable) {
+        List<String> deletedIndicesRouting = new ArrayList<>();
+        for (IndexRoutingTable previousIndexRouting : previousRoutingTable.getIndicesRouting().values()) {
+            if (!currentRoutingTable.getIndicesRouting().containsKey(previousIndexRouting.getIndex().getName())) {
+                // Latest Routing Table does not have entry for the index which means the index is deleted
+                deletedIndicesRouting.add(previousIndexRouting.getIndex().getName());
+            }
+        }
+        return deletedIndicesRouting;
+    }
+
+    public List<String> getIndicesRoutingUpdated(RoutingTable previousRoutingTable, RoutingTable currentRoutingTable) {
```

Comment:
Where are these methods used?

Reply:
I guess this can be removed as @Arpit-Bandejiya is not using DiffUtil to get the changes